### PR TITLE
chore: use plugins for component testing again

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: docs/src/test-components-js.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ npm i @playwright/experimental-ct-react
 ### Create a test `src/App.spec.tsx`
 
 ```js
-import { test, expect } from '@playwright/experimental-ct-react';
+import { test, expect } from '@playwright/test';
```
Comment (rwoll) [inline docs/src/test-components-js.md:65]: I expected to see more changes in this doc mentioning including the plugin and the typings. They'll see it in the `examples/*`, but it's worth pointing out which few lines in there are the wiring.

File: packages/playwright-test/types/test.d.ts [hunk 2]
```diff
@@ -475,7 +474,7 @@ interface TestConfig {
    *
    */
   webServer?: TestConfigWebServer;
-  plugins?: TestPlugin[],
+  plugins?: (TestPlugin | string | [string, any])[],
```
Comment (rwoll) [inline packages/playwright-test/types/test.d.ts:477]: All that nice typing…and then an `any` for the string load case?!

File: packages/playwright-test/types/test.d.ts [hunk 3]
```diff
@@ -2620,7 +2619,7 @@ export type VideoMode = 'off' | 'on' | 'retain-on-failure' | 'on-first-retry';
  * ```
  *
  */
-export interface PlaywrightTestOptions {
+export interface PlaywrightTestOptions extends PlaywrightTest.TestOptions {
```
Comment (rwoll) [inline packages/playwright-test/types/test.d.ts:2622]: This is an odd direction of extension. If user overrides `acceptDownload: string` in their global, what will TS compiler say? I think it will yell about `PlaywrightTestOptions`, but we want it to say `PlaywrightTest.TestOptions` is wrong.

File: tests/components/ct-react/tsconfig.json [hunk 4]
```diff
@@ -21,6 +21,7 @@
     "jsx": "react-jsx"
   },
   "include": [
-    "src"
+    "src",
```
Comment (rwoll) [inline tests/components/ct-react/tsconfig.json:24]: nit: we mix styling in tsconfig.json. Looks like other cases we put everything on one line, and others we do this.

File: utils/generate_types/index.js [hunk 5]
```diff
@@ -93,7 +93,7 @@ class TypesGenerator {
 
     let overrides = await parseOverrides(overridesFile, className => {
       const docClass = this.docClassForName(className);
-      if (!docClass)
+      if (!docClass || !this.shouldGenerate(className))
```
Comment (rwoll) [inline utils/generate_types/index.js:96]: I think this is fine, but double check with @dgozman

File: utils/generate_types/overrides-test.d.ts [hunk 6]
```diff
@@ -57,13 +57,14 @@ export interface FullProject<TestArgs = {}, WorkerArgs = {}> {
 type LiteralUnion<T extends U, U = string> = T | (U & { zz_IGNORE_ME?: never });
 
 export interface TestPlugin {
+  name: string;
```
Comment (rwoll) [inline utils/generate_types/overrides-test.d.ts:60]:
```suggestion
  // [internal] other fields are added via docs
  name: string;
```

**Review by rwoll — APPROVED** [review]
Looks good to land…some nits, and I want to play around more with the typings post-land to see how TS compiler behaves in some edge cases with its error messaging.

File: docs/src/test-components-js.md (hunk 1 above)
Comment (pavelfeldman) [reply docs/src/test-components-js.md:65]: I'll follow up when the story settles

File: packages/playwright-test/types/test.d.ts (hunk 2 above)
Comment (pavelfeldman) [reply packages/playwright-test/types/test.d.ts:477]: I'll follow up with simplification!

File: packages/playwright-test/types/test.d.ts (hunk 3 above)
Comment (pavelfeldman) [reply packages/playwright-test/types/test.d.ts:2622]: Error would still be actionable i think

File: utils/generate_types/index.js (hunk 5 above)
Comment (pavelfeldman) [reply utils/generate_types/index.js:96]: It affected the output in the expected manner, so should be good. We commit the results of the run.

File: utils/generate_types/overrides-test.d.ts (hunk 6 above)
Comment (pavelfeldman) [reply utils/generate_types/overrides-test.d.ts:60]: will follow up
